### DISCUSSION TITLE
Can MsQuic be used in browsers?

### DISCUSSION THREAD

**Discussion by pfrex** [microsoft/msquic#2273] [Q&A] [answered]
I'm interested in using MsQuic to develop multiplayer HTML5 games. Earlier discussions mention HTTP3 and WebTransport, which seem to be built on top of Quic? I found this blog post which mentions successful interop of MsQuic with canary versions of browsers. 

https://techcommunity.microsoft.com/t5/networking-blog/msquic-is-open-source/ba-p/1345441

Could MsQuic be compiled with a tool like Emscripten to run in the browser? 

I think I may be well out of my comfort zone here and will probably just stick with the WebSocket protocol currently but spent too much time looking at this to not post a question.

**Comment by nibanks** [comment, marked as answer]
To work in browsers, you need not only QUIC/MsQuic, but HTTP/3 (and WebTransport). The best solution that leverages QUIC w/ HTTP/3 right now is .NET. We don't currently have any publicly available HTTP/3 solution for MsQuic that is C/C++.
